client/state/ui/comments/reducer.js

```diff
@@ -24,20 +23,42 @@ const deepUpdateComments = ( state, comments, { page = 1, postId, search, status
 			...parentObject,
 			[ filter ]: {
 				...filterObject,
-				[ page ]: commentIds,
+				[ page ]: comments,
 			},
 		},
 	};
 };
 
 export const queries = ( state = {}, action ) => {
-	if ( isUndefined( get( action, 'query.page' ) ) ) {
-		return state;
-	}
-	const { comments, type, query } = action;
-	switch ( type ) {
+	switch ( action.type ) {
+		case COMMENTS_CHANGE_STATUS:
+		case COMMENTS_DELETE:
+			if ( ! action.refreshCommentListQuery ) {
+				return state;
+			}
+			const { page, postId, search, status } = action.refreshCommentListQuery;
+			if (
+				COMMENTS_DELETE !== action.type &&
```

Comment:
How about checking for `COMMENTS_CHANGE_STATUS`?

Reply:
Maybe leave a quick comment that we no-op when status changes from approved or unapproved in the all tab.